asmcli/lib/util.sh

```diff
@@ -42,6 +42,28 @@ run_command() {
   return $RETVAL
 }
 
+######
+# check_curl calls curl and returns non-zero if the http code is not 200 or the
+# curl command fails.
+######
+check_curl() {
+  local TMPFILE; TMPFILE=$(mktemp)
+  local HTTPCODE; HTTPCODE=$(run_command curl --write-out '%{http_code}' --silent --show-error --output "$TMPFILE" "${@}")
+  local RETVAL; RETVAL="$?"
```

Comment:
If you do it like this then `$?` is going to have the return code from `local RETVAL` and not `curl`. I probably just checking if $HTTPCODE is empty might be what you want.

Reply:
You want to check both because if curl fails, HTTPCODE will be "000", which is a bit confusing, so it's probably better to just pass along the curl error code instead. Moved the `local` calls to above the curl command.